.github/workflows/codeql.yml

```diff
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", hotfix-[0-9]*.[0-9]*.[0-9]*, release-* ]
```

Comment:
Do we need to define the branch patterns? I'm not happy with our current wildcard patterns and plan to rework them at some point

Can we run it on all branches? Skip the arg or use `*`?